authentication/context.go

```diff
@@ -0,0 +1,28 @@
+package authentication
+
+import (
+	"context"
+	"errors"
+)
+
+type ctxKey int
+
+// CtxKey -
+const CtxKey ctxKey = 0
+
+// ErrNoCurrentUserInCtx -
+var ErrNoCurrentUserInCtx = errors.New("no current user in context")
+
+// GetCurrentUserFromCtx -
+func GetCurrentUserFromCtx(ctx context.Context) *CurrentUser {
```

Comment:
Go's packages are designed to add meaning to the functions that they make public. Therefore, `authentication.GetFromContext(ctx)` means "get the authentication from the context ctx", whereas in `authentication.GetCurrentUserFromCtx(ctx)`, `authentication` becomes simply another namespace required to call a function.